### PR TITLE
#6Task. Mikhail Pyshnik

### PR DESCRIPTION
**Pull request by MikhailPyshnik**
https://travis-ci.org/github/MikhailPyshnik/js-assignments/builds/704886052

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,10 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+    if (num%15==0) return 'FizzBuzz';
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: It is preferable to use operator "===".
Need to use indents between operators.
It is necessary to follow the generally accepted coding style.
https://javascript.info/coding-style

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -46,7 +49,12 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
-    throw new Error('Not implemented');
+    let sumOfFactorial = 1;
+    for(let i = 1; i <= n; i ++)
+    {
+        sumOfFactorial*=i;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Need to use indents between operators.

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -63,7 +71,12 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+    let sum = 0;
+    for( let i = n1; i<=n2; i ++)
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Need to use indents between operators.
It is necessary to correctly indent the brackets.
https://javascript.info/coding-style

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -119,7 +139,13 @@ function isTriangle(a,b,c) {
  *  
  */
 function doRectanglesOverlap(rect1, rect2) {
-    throw new Error('Not implemented');
+    if(rect1.top < rect2.top + rect2.width && rect2.top < rect1.top + rect1.width &&
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js:138]: You need to return the result of a boolean expression.
The "if, else if" construction is redundant.

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -296,7 +394,38 @@ function getDigitalRoot(num) {
...
+        {
+            openIndex = arrTemp.pop();
+            if (closeIndex !== openIndex) 
+            {
+                return false;
+            }
+        }
+    }
+
+    if (arrTemp.length != 0)
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: Here you can return the result of a boolean expression.

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -270,7 +350,25 @@ function isCreditCardNumber(ccn) {
...
+        {
+            sumDigitals += numbers % 10;
+            numbers = (numbers - numbers % 10)/10;
+        }
+
+        if(sumDigitals > 9) 
+        {
+            return countSumOfDigitals(sumDigitals);
+        }
+        else return sumDigitals;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: "else" is redundant

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -270,7 +350,25 @@ function isCreditCardNumber(ccn) {
  *   165536 (1+6+5+5+3+6 = 26,  2+6 = 8) => 8
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    let result = countSumOfDigitals(num);
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js:320]: Here you can return the result of a function without using variable or even do without a nested function.

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -82,7 +95,14 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if(a+b>c && a+c>b && b+c>a)
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: You can return just boolean expression result.
Need to use indents between operators.

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -192,7 +227,26 @@ function findFirstSingleChar(str) {
  *
  */
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+    let resultString = "";
+    let smallerNumber = 0;
+    
+     a > b ? (smallerNumber = b) : (smallerNumber = a , a = b);
+
+    return resultString =`${chekingBrackets(isStartIncluded,"open")}${smallerNumber}, ${a}${chekingBrackets(isEndIncluded,"cloced")}`;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: This is too complex a solution for this simple task.

File: task/06-conditions-n-loops-tasks.js [hunk 10]
```diff
@@ -209,7 +263,12 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
-    throw new Error('Not implemented');
+    let resultReverseString = '';
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: For this task, you can also use a simpler solution through the "reverse" method

File: task/06-conditions-n-loops-tasks.js [hunk 11]
```diff
@@ -226,7 +285,13 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    let resultReversInteger = 0;
```
Comment (KostyaKov) [inline task/06-conditions-n-loops-tasks.js]: For this task, you can also use a simpler solution through the "reverse" method, and converting to string and back.

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js (hunk 11 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 10 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 9 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 8 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 7 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js:320]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 6 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 5 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 4 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js:138]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 3 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 2 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js (hunk 1 above)
Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed.

File: task/06-conditions-n-loops-tasks.js [hunk 12]
```diff
@@ -419,13 +386,7 @@ function isBracketsBalanced(str) {
...
         }
     }
-
-    if (arrTemp.length != 0) 
-    {
-        return false;
-    }
-
-    return true;
+    return (arrTemp.length == 0);
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable
sometimes ''!=' may cause an error
(check at console :
"0" == 0
and "0" === 0)

Comment (MikhailPyshnik) [reply task/06-conditions-n-loops-tasks.js]: Fixed comment.

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**